pkg/cloud/services/compute/instance.go

```diff
@@ -630,21 +641,25 @@ func (s *Service) DeleteInstance(object runtime.Object, instanceName string) err
 
 		// delete port
 		err = util.PollImmediate(RetryIntervalPortDelete, TimeoutPortDelete, func() (bool, error) {
+			mc := metrics.NewMetricPrometheusContext("port", "delete")
 			err := ports.Delete(s.networkClient, port.PortID).ExtractErr()
 			if err != nil {
```

Comment:
```suggestion
			if mc.ObserveRequest(err) != nil {
```
instead of the two mc.ObserveRequest below